src/builtins.ts

```diff
@@ -1484,7 +1485,7 @@ function builtin_max(ctx: BuiltinContext): ExpressionRef {
     if (op != -1) {
       let flow = compiler.currentFlow;
       let nativeType = type.toNativeType();
-      let temp1 = flow.getTempLocal(type);
+      let temp1 = flow.getTempLocal(type, findUsedLocals(arg1));
```

Comment:
how about make `getUnusedLocal` which try to find unused allocated locals or allocate new one otherwise? And use `flow.getUnusedLocal(type)` here and in other places?

Reply:
`findUsedLocals` is what other code typically uses to keep the number of necessary locals low. Otherwise we'd always get a new local, never a reused one, I guess?

Reply:
Also `builtin_abs`, `builtin_rotl` / `builtin_rotr` I guess also required this fix

Reply:
Btw, at some point I expect us to move away from free'ing temporary locals, but we'll have to get rid of any kind of recompiling expressions first. There are still a few places doing this, and `getUnusedLocal` would return a new local upon recompile, yielding different code, while `findUsedLocals` doesn't.

Reply:
> Also builtin_abs, builtin_rotl / builtin_rotr I guess also required this fix

Abs is a unary expression not executing input code in between assignment and usage of the temporary locals. Should be safe. Going to take a look at rotl/rotr.

Reply:
> findUsedLocals is what other code typically uses to keep the number of necessary locals low. Otherwise we'd always get a new local, never a reused one, I guess?

I guess better always alloc new loacal and let binaryen optimize all of this afterwards

Reply:
I expect us to move into that direction, yeah, but we cannot do that as long as we are recompiling code because it would confuse local flags (adding new locals on each recompile, leaving earlier ones unused), then for example leading to infinite loops when attempting to unify local flags in loops.

Reply:
Seems that `rotl` for some reason does not exhibit the behavior on `rotl(a, rotl(b, c))`, but should be susceptible just as `rotr` is, which trivially breaks without the fix. Probably coincidence. Applied the same fix for both.

Reply:
> then for example leading to infinite loops when attempting to unify local flags in loops.

I see. Ok, in this case LGTM